src/server/survey_and_program_data.R

```diff
@@ -68,14 +68,11 @@ validateProgramData <- function(df) {
 
     validateYear <- function(givenYear) {
 
-        rows <- df[!is.na(df$year) & df$year == givenYear,]
+        rows <- df[which(df$year == givenYear),]
 
-        if (nrow(rows) == 0) return(TRUE)
-        if (nrow(rows) > 2) return(FALSE)
-        if (nrow(rows) == 1 & identical(as.character(rows$sex),c("both"))) return(TRUE)
-        if (nrow(rows) == 2 & identical(sort(as.character(rows$sex)), sort(c("male", "female")))) return(TRUE)
-
-        FALSE
+        nrow(rows) == 0 |
+        identical(as.character(rows$sex),c("both")) |
```

Comment:
realised this implicitly checks the number of rows, so no need to check `nrow` explicitly

Reply:
Very nice!